lsoa/views.py

```diff
@@ -458,3 +459,55 @@ def get_recent_observations(owner, age=None):
     age = age or timedelta(hours=2)
     within_timeframe = timezone.now() - age
     return Observation.objects.filter(owner=owner, created__gte=within_timeframe).order_by('-created')
+
+
+class BaseTagManagement(LoginRequiredMixin):
+    model = ContextTag
+    form_class = ContextTagForm
+    success_url = reverse_lazy('setup')
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['title'] = self.title
+        return context
+
+
+class CreateTag(BaseTagManagement, CreateView):
+    template_name = 'context_tag.html'
+    title = 'Create Tag'
+
+    def form_valid(self, form):
+        self.object = form.save(commit=False)
+        self.object.owner = self.request.user
+        self.object.save()
+        return super().form_valid(form)
+
+
+class EditTag(BaseTagManagement, UpdateView):
+    template_name = 'context_tag.html'
+    title = 'Edit Tag'
+
+    def check_owner(self, request, pk):
```

Comment:
Instead of writing custom checks, you can limit available tags by using get_queryset to return only tags owned by current user, and it'll handle 403 automatically.